src/Processors/Transforms/Streaming/AggregatingTransformWithSubstream.cpp

```diff
@@ -226,10 +226,10 @@ std::pair<bool, bool> AggregatingTransformWithSubstream::executeOrMergeColumns(C
     /// according to partition keys
     auto num_rows = chunk.getNumRows();
 
-    assert(!params->only_merge);
+    assert(!params->only_merge && !no_more_keys);
```

Comment:
no_more_keys ? We can get rid of it in separate PR if necessary since this one is big enough